src/operations.cpp

```diff
@@ -1399,6 +1419,67 @@ void last_write_time(const path& p, const std::time_t new_time,
 # endif
 }
 
+BOOST_FILESYSTEM_DECL
+std::time_t creation_time(const path& p, system::error_code* ec)
+{
+# ifdef BOOST_POSIX_API
+# ifdef __linux__
+  struct statx path_stat;
+  if (error(::statx(AT_FDCWD, p.c_str(), AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT |
+	AT_STATX_SYNC_AS_STAT, STATX_BTIME, &path_stat)!= 0 ? BOOST_ERRNO : 0,
+    p, ec, "boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.stx_btime.tv_sec);
+# elif defined(__APPLE__)
+  // The stat64() system calls first appeared in Mac OS X 10.5 (Leopard)
+#   if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
+#     ifdef _DARWIN_USE_64_BIT_INODE
```

Comment:
This is incorrect, as this is the macro you define to request the 64-bit inodes. Whether the feature is enabled is indicated by `_DARWIN_FEATURE_64_BIT_INODE`. http://polarhome.com/service/man/?qf=stat&tf=2&of=Darwin&sf=2